pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go

```diff
@@ -81,3 +159,90 @@ func Test_COSConfig(t *testing.T) {
 		require.Equal(t, []string{tt.cosConfig.BucketNames}, cosClient.bucketNames)
 	}
 }
+
+func Test_DeleteObject(t *testing.T) {
+	tests := []struct {
+		key     string
+		wantErr error
+	}{
+		{
+			"key-4",
+			nil,
+		},
+		{
+			"key-5",
+			errMissingObject,
+		},
+	}
+
+	for _, tt := range tests {
+		cosConfig := COSConfig{
+			BucketNames:     bucket,
+			Endpoint:        "test",
+			Region:          "dummy",
+			AccessKeyID:     "dummy",
+			SecretAccessKey: flagext.SecretWithValue("dummy"),
+			BackoffConfig: backoff.Config{
+				MaxRetries: 1,
+			},
+		}
+		cosClient, err := NewCOSObjectClient(cosConfig, hedging.Config{})
+		require.NoError(t, err)
+		cosClient.cos = newMockCosClient()
+		err = cosClient.DeleteObject(context.Background(), tt.key)
+		if tt.wantErr != nil {
+			require.Equal(t, tt.wantErr.Error(), err.Error())
+			continue
+		}
+		require.NoError(t, err)
```

Comment:
Could you please add a get call and check the deleted object is no longer exist in bucket.